internal/cmd/request.go

```diff
@@ -0,0 +1,107 @@
+package cmd
+
+import (
+	"fmt"
+	"net"
+	"reflect"
+
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/api/server"
+	"github.com/edulinq/autograder/internal/common"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+type CommonOptions struct {
+	Verbose bool `help:"Use verbose output to show full request/response without specific formatting." default:"false"`
+}
+
+func MustHandleCMDRequestAndExit(endpoint string, request any, responseType any) error {
+	return MustHandleCMDRequestAndExitFull(endpoint, request, responseType, CommonOptions{}, nil)
+}
+
+func MustHandleCMDRequestAndExitFull(endpoint string, request any, responseType any, options CommonOptions, customPrintFunc any) error {
```

Comment:
This print function should probably be a declared type.
E.g.:
https://github.com/edulinq/autograder-server/blob/main/internal/api/core/routing.go#L21